public/ledger.js

```diff
@@ -154,62 +154,64 @@ class Ledger {
     this.sendQueue = [];
 
     // If can send command to ledger
-    this.canSend = true;
+    this.runningQueue = false;
   }
 
   /**
    * Send a command to ledger if there's anything on the send queue.
+   * If the queue is empty, close the transport.
    */
   checkSendQueue = () => {
-    if (this.sendQueue.length > 0) {
-      const { transport, command, p1, p2, data, resolve, reject } = this.sendQueue.shift();
-      const promise = transport.send(ledgerCLA, command, p1, p2, data);
-      promise.then((response) => {
-        resolve(response);
-      }, (error) => {
-        reject(error);
-      }).finally(() => {
-        this.checkSendQueue();
-      });
-    } else {
-      // no more commands on queue, we can release the transport connection
-      if (this.transport) {
+    if (this.sendQueue.length === 0) {
+      console.log('closing transport!');
+      this.runningQueue = false;
+      // queue is exausted, close transport and stop
+      if(this.transport) {
         this.transport.device.close();
         this.transport = null;
       }
-      this.canSend = true;
+      return;
     }
+
+    const { command, p1, p2, data, resolve, reject } = this.sendQueue.shift();
+    const promise = this.getTransport().then(transport => transport.send(ledgerCLA, command, p1, p2, data));
+    promise.then((response) => {
+      resolve(response);
+    }, (error) => {
+      reject(error);
+    }).finally(() => {
+      // When done, check the queue again
+      this.checkSendQueue();
+    });
```

Comment:
As a code style suggestion:
```suggestion
const { command, p1, p2, data, resolve, reject } = this.sendQueue.shift();
    this.getTransport()
      .then((transport) => transport.send(ledgerCLA, command, p1, p2, data))
      .then((response) => resolve(response), (error) => reject(error))
      .finally(() => {
        // When done, check the queue again
        this.checkSendQueue();
      });
```